src/Entities/Player.cpp

```diff
@@ -941,10 +941,32 @@ void cPlayer::ApplyArmorDamage(int DamageBlocked)
 	{
 		ArmorDamage = 1;
 	}
-	m_Inventory.DamageItem(cInventory::invArmorOffset + 0, ArmorDamage);
-	m_Inventory.DamageItem(cInventory::invArmorOffset + 1, ArmorDamage);
-	m_Inventory.DamageItem(cInventory::invArmorOffset + 2, ArmorDamage);
-	m_Inventory.DamageItem(cInventory::invArmorOffset + 3, ArmorDamage);
+
+	for (int i = 0; i < 4; i++)
+	{
+		// (60 + (40 / (UnbreakingLevel + 1)))% chance the armor takes damage
+		// Ref: https://minecraft.gamepedia.com/Enchanting#Unbreaking
+		cItem ArmorPiece = m_Inventory.GetArmorSlot(i);
+		if (ArmorPiece.IsEmpty())
+		{
+			continue;
+		}
+		unsigned int UnbreakingLevel = ArmorPiece.m_Enchantments.GetLevel(cEnchantments::enchUnbreaking);
+		double chance = (0.6 + (0.4 / (UnbreakingLevel + 1)));
+
+		// When durability is reduced by multiple points
+		// Unbreaking is applied for each point of reduction.
+		std::binomial_distribution<short> Dist(ArmorDamage, chance);
+		auto ReducedDamage = Dist(GetRandomProvider().Engine());
+		if (ReducedDamage == 0)
+		{
+			continue;
+		}
+		if (m_Inventory.DamageItem(cInventory::invArmorOffset + i, 1))
+		{
+			m_World->BroadcastSoundEffect("entity.item.break", GetPosition(), 0.5f, static_cast<float>(0.75 + (static_cast<float>((GetUniqueID() * 23) % 32)) / 64));
```

Comment:
These calculations should definitely go into a seperate function. And the `0.75` + `GetUniqueID()` calculation should be explained (at least by using a function name like `CalculateSoundVolume`).

Reply:
I just copied this from here: https://github.com/cuberite/cuberite/blob/master/src/Entities/Player.cpp#L2349
for consistency.